client/my-sites/themes/theme-showcase.jsx

```diff
@@ -107,10 +107,24 @@ class ThemeShowcase extends React.Component {
 	};
 
 	componentDidMount() {
+		const { search, filter, hasShowcaseOpened, themesBookmark } = this.props;
 		// Open showcase on state if we open here with query override.
-		if ( ( this.props.search || this.props.filter ) && ! this.props.hasShowcaseOpened ) {
+		if ( ( search || filter ) && ! hasShowcaseOpened ) {
 			this.props.openThemesShowcase();
 		}
+		// Scroll to bookmark if applicable.
+		if ( themesBookmark ) {
+			setTimeout( () => {
+				const lastTheme = this.bookmarkRef.current;
+				if ( lastTheme ) {
+					lastTheme.scrollIntoView( {
+						behavior: 'auto',
+						block: 'center',
+						inline: 'center',
+					} );
+				}
+			}, 40 );
```

Comment:
@noahtallen what are your thoughts on the timeout?

I found without the timeout it would always return to the top of the page.  After digging around the code and testing different `scrollTo( 0, 0 )`s that were in place, I was not successful in finding anything as the specific cause.  Adding the timeout seems to make it work correctly, but I am still not too happy about having some arbitrary number of ms sitting there as the fix.

I could make this `10` instead of `40` and it still seems to work fine.  Will this have issues on slower machines?  or will it not matter since the timeout will always run after whatever instantaneous post-mount function is causing this odd behavior?  I am inclined to believe the latter but I figured it would be worth pointing this out.

Reply:
The only thing I can think of is weird issues with the JS event loop. Since we have a timeout set, it defers the callback until the next iteration of the event loop -- i.e. it does not happen synchronously. Perhaps something in the immediate execution needs to happen before the scroll will work. If that's the case, then you should be able to use `setTimeout` of 0 in order to get JS to run the callback asynchronously

Reply:
If you haven't heard about this (I hadn't either until a few months ago 😁), you can read up on it here: https://stackoverflow.com/questions/779379/why-is-settimeoutfn-0-sometimes-useful (there are some useful videos linked too)

Reply:
Awesome, thanks for the input and info.  That makes sense and it works well with no delay on the timeout, the scroll just needed to be at the end of the queue for whatever reason.  I have updated it to just be `setTimeout( func )` instead which makes me feel a bit better. 😄

Reply:
Great!